### PR TITLE
Use Guava's Multimap for test server headers

### PR DESCRIPTION
**Pull request by mar-kolya**

### CODE REVIEW

File: dd-java-agent/testing/src/main/groovy/datadog/trace/agent/test/server/http/HttpServletRequestExtractAdapter.java [hunk 1]
```diff
@@ -1,95 +1,50 @@
...
-import java.util.List;
 import java.util.Map;
-import java.util.Set;
 import javax.servlet.http.HttpServletRequest;
 
 /**
  * Tracer extract adapter for {@link HttpServletRequest}.
  *
- * @author Pavol Loffay
+ * <p>FIXME: we have 4 copies of this class now - we should really make this a utility or something
```
Comment (gary-huang) [inline dd-java-agent/testing/src/main/groovy/datadog/trace/agent/test/server/http/HttpServletRequestExtractAdapter.java]: I agree with this, but this implementation that uses Guava should only be used in test source sets right? The real implementation used in instrumentations should be something that doesn't use Guava's multimap.

Comment (mar-kolya) [reply dd-java-agent/testing/src/main/groovy/datadog/trace/agent/test/server/http/HttpServletRequestExtractAdapter.java]: I personally would prefer to 'bite the bullet' and figure out how to repackage guava under DD namespace - we already do this for other packages, so this should be solvable problem... And then we would not have to reimplement things :)

**Comment by tylerbenson** [comment]
I'd rather not make the adapter different and I don't think the duplication is a huge deal, though I agree we could reduce the duplication by extracting it to a library.  I applied your code coverage changes and improved coverage some myself.  Mind if we close this?

**Comment by mar-kolya** [comment]
I would appreciate at least a FIXME in original code to outline that we eventually need to address code duplication.
But I guess we can close this.
Thanks!

**Pull request closed without merging**